src/PlatformEngines.jl

```diff
@@ -686,6 +690,107 @@ function get_auth_header(url::AbstractString; verbose::Bool = false)
     return "Authorization: Bearer $(auth_info["access_token"])"
 end
 
+function hash_data(strs::AbstractString...)
+    ctx = SHA.SHA224_CTX()
+    for str in strs
+        data = Vector{UInt8}(str)
+        len = length(data)
+        while true
+            push!(data, len % UInt8)
+            len == 0 && break
+            len >>= 8
+        end
+        SHA.update!(ctx, data)
+    end
+    return bytes2hex(@view SHA.digest!(ctx)[1:20])
+end
+
+function load_telemetry_file(file::AbstractString)
+    if !ispath(file)
+        info, changed = Dict(), true
+    else
+        info, changed = try
+            TOML.parsefile(file), false
+        catch err
+            @warn "replacing malformed telemetry file" file=file err=err
+            Dict(), true
+        end
+    end
+    # bail early if fully opted out
+    get(info, "telemetry", true) == false && return info
+    # some validity checking helpers
+    is_valid_uuid(x) = false
+    is_valid_salt(x) = false
+    is_valid_uuid(x::Bool) = !x # false is valid, true is not
+    is_valid_salt(x::Bool) = !x # false is valid, true is not
+    is_valid_uuid(x::AbstractString) = occursin(Pkg.REPLMode.uuid_re, x)
+    is_valid_salt(x::AbstractString) = occursin(r"^[0-9a-zA-Z]+$", x)
+    # generate or fix system-specific info
+    if !haskey(info, "client_uuid") || !is_valid_uuid(info["client_uuid"])
+        info["client_uuid"] = string(uuid4())
+        changed = true
+    end
+    if !haskey(info, "secret_salt") || !is_valid_salt(info["secret_salt"])
+        info["secret_salt"] = randstring(36)
+        changed = true
+    end
+    if changed
+        mkpath(dirname(file))
+        open(file, write=true) do io
+            TOML.print(io, info, sorted=true)
+        end
+    end
+    return info
+end
+
+const CI_VARS = [
+    "APPVEYOR",
+    "CI",
+    "CIRCLECI",
+    "CONTINUOUS_INTEGRATION",
+    "GITHUB_ACTION",
```

Comment:
This comment comes quite too late... Actually IIUC it should be GITHUB_ACTIONS (note tailing S here) according to [default GitHub action environment variables](https://help.github.com/en/actions/configuring-and-managing-workflows/using-environment-variables#default-environment-variables)

And `GITHUB_ACTION` is the job id

Reply:
Thanks, will fix this.

Reply:
PR: https://github.com/JuliaLang/Pkg.jl/pull/1693